### DISCUSSION TITLE
Generated code is not marked with GeneratedCodeAttribute

### DISCUSSION THREAD

**Discussion by albanur** [CommunityToolkit/Maui#1467] [New Feature Discussions]
The code generated by CommunityToolkit.Maui.SourceGenerators.Generators.TextColorToGenerator is not marked with [GeneratedCodeAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.codedom.compiler.generatedcodeattribute?view=net-7.0). Generated code is not clearly marked as generated code and can be treated like a code written manually by developers. For example this causes problem with code coverage tools/frameworks, which shows generated code in reports as a code not covered with unit tests.

Another solution is to mark generated code with [ExcludeFromCodeCoverageAttribute](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.excludefromcodecoverageattribute?view=net-7.0), however the first option is more precise.

**Comment by TheCodeTraveler** [comment]
Thanks @albanur. 

Would you like to submit a Pull Request?

**Reply by TheCodeTraveler** [reply]
@albanur I have opened a Pull Request adding these attributes here: https://github.com/CommunityToolkit/Maui/pull/1468

When you get a chance, could you please review / approve the PR?

**Reply by albanur** [reply]
Thanks @brminnick for quick response. Unfortunately it happened to fast and I was not able to submit a Pull Request nor review your.

**Comment by pictos** [comment]
PRs merged.
